tests/core/data_model/test_dataobject.py

```diff
@@ -575,6 +634,101 @@ class Foo(DataObjectBase):
     assert Foo.from_json(json_repr_foo) == foo1
 
 
+def test_dataobject_list_oneof_round_trips_with_specified_names():
+    @dataobject
+    class Foo(DataObjectBase):
+        foo: Union[
+            Annotated[List[int], FieldNumber(10), OneofField("myfoointseq")],
+            Annotated[List[str], FieldNumber(20), OneofField("myfoostrseq")],
```

Comment:
quick question: What's different between this test and the one above that has
```
    class Foo(DataObjectBase):
        foo: Union[
            Annotated[int, FieldNumber(10), OneofField("myfooint")],
            Annotated[float, FieldNumber(20), OneofField("myfoofloat")],
        ]
```

I can't quickly tell if the only difference here is the names or if there's something else different going on

Reply:
This one is union of lists, the above is union of int and float